app/plugin_api.go

```diff
@@ -694,6 +694,10 @@ func (api *PluginAPI) GetPluginStatus(id string) (*model.PluginStatus, *model.Ap
 	return api.app.GetPluginStatus(id)
 }
 
+func (api *PluginAPI) GetPluginStatuses() (model.PluginStatuses, *model.AppError) {
```

Comment:
In order for this to be exposed to plugins, it's necessary to add it to `plugin/api.go` and regenerate the client code like the hooks. Unit tests, similar to the hooks below, will help make this clearer.